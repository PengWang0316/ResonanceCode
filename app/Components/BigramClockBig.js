import React from 'react';
import PropTypes from 'prop-types';

import styles from '../styles/BigramClockBig.module.css';
import needleBig from '../imgs/needleBig.png';
import growthImage from '../imgs/growth.png';
import conceptionImage from '../imgs/conception.png';
import seedingImage from '../imgs/seeding.png';
import maturationImage from '../imgs/maturation.png';

const imgStyles = {
  0: styles.needle0,
  1: styles.needle1,
  2: styles.needle2,
  3: styles.needle3
};

const BigramClockBig = ({ lineText, position }) => (
  <div className={styles.mainDiv}>
    <div className={styles.lineText}>{lineText}</div>
    <div className="text-center">
      <div className={styles.bigramName}>Potentiation</div>
      <div><img src={conceptionImage} alt="Potentiation" /></div>

      <div className="d-flex justify-content-center">
        <div className={styles.circleBig}>
          <div className={styles.growthDiv}>
            <div><img src={growthImage} alt="Growth" /></div>
            <div className={styles.bigramName}>Growth</div>
          </div>
          <div className={styles.reSourcingDiv}>
            <div><img src={seedingImage} alt="Re-Sourcing" /></div>
            <div className={styles.bigramName}>Re-Sourcing</div>
          </div>
          <img src={needleBig} alt="needle" className={imgStyles[position]} />
        </div>
      </div>

      <div><img src={maturationImage} alt="Maturation" /></div>
      <div className={styles.bigramName}>Maturation</div>

    </div>
  </div>
);
BigramClockBig.propTypes = {
  lineText: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired
};
export default BigramClockBig;
