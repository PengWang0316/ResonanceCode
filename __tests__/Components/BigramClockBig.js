import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

import BigramClockBig from '../../app/Components/BigramClockBig';

describe('BigramClockBig test', () => {
  const defaultProps = {
    lineText: 'test text',
    position: 0
  };
  // const getShallowComponent = (props = defaultProps) => shallow(<BigramClockBig {...props} />);

  test('BigramClockBig snapshot', () => expect(renderer.create(<BigramClockBig {...defaultProps} />).toJSON()).toMatchSnapshot());
});
