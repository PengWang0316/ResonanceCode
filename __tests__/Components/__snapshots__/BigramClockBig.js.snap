// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigramClockBig test BigramClockBig snapshot 1`] = `
<div
  className="mainDiv"
>
  <div
    className="lineText"
  >
    test text
  </div>
  <div
    className="text-center"
  >
    <div
      className="bigramName"
    >
      Potentiation
    </div>
    <div>
      <img
        alt="Potentiation"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="d-flex justify-content-center"
    >
      <div
        className="circleBig"
      >
        <div
          className="reSourcingDiv"
        >
          <div>
            <img
              alt="Re-Sourcing"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </div>
          <div
            className="bigramName"
          >
            Re-Sourcing
          </div>
        </div>
        <div
          className="growthDiv"
        >
          <div>
            <img
              alt="Growth"
              src={
                Object {
                  "process": [Function],
                }
              }
            />
          </div>
          <div
            className="bigramName"
          >
            Growth
          </div>
        </div>
        <img
          alt="needle"
          className="needle0"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </div>
    </div>
    <div>
      <img
        alt="Maturation"
        src={
          Object {
            "process": [Function],
          }
        }
      />
    </div>
    <div
      className="bigramName"
    >
      Maturation
    </div>
  </div>
</div>
`;
